mongodbatlas/data_source_mongodbatlas_serverless_instances.go

```diff
@@ -0,0 +1,86 @@
+package mongodbatlas
+
+import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	matlas "go.mongodb.org/atlas/mongodbatlas"
+)
+
+func dataSourceMongoDBAtlasServerlessInstances() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: dataSourceMongoDBAtlasServerlessInstancesRead,
+		Schema: map[string]*schema.Schema{
+			"project_id": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"results": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: returnServerlessInstanceDSSchema(),
+				},
+			},
+		},
+	}
+}
+
+func dataSourceMongoDBAtlasServerlessInstancesRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	// Get client connection.
+	conn := meta.(*MongoDBClient).Atlas
+
+	projectID, projectIDOK := d.GetOk("project_id")
+
+	if !(projectIDOK) {
+		return diag.Errorf("project_id must be configured")
+	}
+
+	options := &matlas.ListOptions{}
+
+	serverlessInstances, _, err := conn.ServerlessInstances.List(ctx, projectID.(string), options)
```

Comment:
[q] why not nil?
```suggestion
	serverlessInstances, _, err := conn.ServerlessInstances.List(ctx, projectID.(string), nil)
```
also are we worry about users with more instances than the default pagination limit?

Reply:
I already managed the pagination with a recursive function to get all the instances